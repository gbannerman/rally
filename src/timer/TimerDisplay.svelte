<script lang="ts">
  import { convertMillisecondsToTimerValue } from "./timer";

  export let remainingMilliseconds;
</script>

<div class="TimerDisplay">
  <div class="timer">
    {convertMillisecondsToTimerValue(remainingMilliseconds)}
  </div>
</div>

<style>
  .TimerDisplay {
    width: 100px;
    height: 40px;
    text-align: center;
  }

  .timer {
    position: absolute;
    width: 100%;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    background-color: #e9e9e9;
    height: 40px;
    line-height: 20px;
    padding: 10px 10px;
    color: #606c6e;
    font-size: 30px;
    outline: 0 solid transparent;
    border: 0 solid transparent;
    border-radius: 4px;
    box-sizing: border-box;
  }
</style>
