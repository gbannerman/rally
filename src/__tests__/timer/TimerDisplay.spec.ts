import {render} from '@testing-library/svelte'

import TimerDisplay from '../../timer/TimerDisplay.svelte';

describe("TimerDisplay component", () => {
  it("renders correctly", () => {
    const { container } = render(TimerDisplay, {
      remainingMilliseconds: 9_000
    });
    
    expect(container).toMatchSnapshot();
  });
});
