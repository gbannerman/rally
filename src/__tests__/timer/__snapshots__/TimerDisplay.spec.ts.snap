// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerDisplay component renders correctly 1`] = `
<body>
  <div>
    <div
      class="TimerDisplay svelte-c1o0ac"
    >
      <div
        class="timer svelte-c1o0ac"
      >
        00:09
      </div>
    </div>
  </div>
</body>
`;
